swap_meet/vendor.py

```diff
@@ -1,2 +1,76 @@
 class Vendor:
-    pass
\ No newline at end of file
+
+    def __init__(self, inventory=None):
+        self.inventory = inventory if inventory is not None else []
+
+    def add(self, item):
+        self.inventory.append(item) # This method adds the item to the inventory
+        return item
+    
+    def remove(self, item):
+        if item not in self.inventory: # If there is no matching item in the inventory, the method should return False
+            return False 
+        
+        self.inventory.remove(item) # This method removes the matching item from the inventory
+        return item
+
+    def get_by_id(self, id):
+        for item in self.inventory: # Loops through each item in inventory
+            if id == item.id: # Checks for id
+                return item
+        
+        return None
+    
+    def swap_items(self, other_vendor, my_item, their_item):
+        
+        
+        if self.inventory == [] or other_vendor.inventory == []: # Check for empty inventory for user or other vendor
+            return False
+        
+        elif my_item not in other_vendor.inventory and their_item not in self.inventory:
+            self.remove(my_item)
+            self.add(their_item)
+            other_vendor.remove(their_item)
+            other_vendor.add(my_item)
+            return True
+
+    # Instances of Vendor have an instance method named swap_first_item
+    # It takes one argument: an instance of another Vendor (other_vendor), 
+    # representing the friend that the vendor is swapping with
+    def swap_first_item(self, other_vendor):
+
+        if self.inventory == [] or other_vendor.inventory == []: #  Check for empty inventory for user or other vendor
+            return False
+        
+        # Helper variables
+        my_first_item = self.inventory[0]
+        their_first_item = other_vendor.inventory[0]
+                
```

Comment:
I love seeing you all use helper variables and functions, it makes your code so readable, doesn't it?! Great job, y'all!